Content.Server/Connection/ConnectionManager.cs

```diff
@@ -175,26 +207,191 @@ private async Task NetMgrOnConnecting(NetConnectingArgs e)
+                        {
+                            matched = false;
+                        }
+                        else
+                        {
+                            matched = tracker.TimeSpent.TotalMinutes < conditionPlaytime.MinimumPlaytime;
+                        }
+                        denyMessage = Loc.GetString("whitelist-playtime", ("minutes", conditionPlaytime.MinimumPlaytime));
+                        break;
+                    case { } t when t == typeof(ConditionNotesPlaytimeRange):
+                        var conditionNotesPlaytimeRange = (ConditionNotesPlaytimeRange)condition;
+                        var remarksPlaytimeRange = await GetAdminRemarks(data.UserId);
+                        // In order to filter by playtime, we need to do the following: playtimeAtNote >= overallPlaytime - Range
+                        var overallPlaytime = await _db.GetPlayTimes(data.UserId);
+                        var overallTracker = overallPlaytime.Find(p => p.Tracker == PlayTimeTrackingShared.TrackerOverall);
+                        if (overallTracker is null)
+                        {
+                            matched = false;
+                            denyMessage = Loc.GetString("whitelist-notes");
+                            break;
+                        }
+                        remarksPlaytimeRange = remarksPlaytimeRange.Where(x =>
+                            x.PlaytimeAtNote >= overallTracker.TimeSpent - TimeSpan.FromMinutes(conditionNotesPlaytimeRange.Range)).ToList();
+                        if (!conditionNotesPlaytimeRange.IncludeExpired)
+                            remarksPlaytimeRange = remarksPlaytimeRange.Where(x => x.ExpirationTime is null || x.ExpirationTime > DateTime.Now).ToList();
+                        var copy = remarksPlaytimeRange.ToList();
+                        foreach (var adminRemarksRecord in remarksPlaytimeRange)
+                        {
+                            // In order to get the severity of the remark, we need to see if its a AdminNoteRecord.
+                            if (adminRemarksRecord is not AdminNoteRecord adminNoteRecord)
+                                continue;
+
+                            // We want to filter out secret notes if we're not including them.
+                            if (!conditionNotesPlaytimeRange.IncludeSecret && adminNoteRecord.Secret)
+                            {
+                                copy.Remove(adminRemarksRecord);
+                                continue;
+                            }
+
+                            // At this point, we need to remove the note if it's not within the severity range.
+                            if (adminNoteRecord.Severity < conditionNotesPlaytimeRange.MinimumSeverity)
+                            {
+                                copy.Remove(adminRemarksRecord);
+                                continue;
+                            }
+                        }
```

Comment:
Same concerns with this code being half LINQ half not. Also isn't this copy pasted?